kore/src/Kore/Step/Simplification/Conditional.hs

```diff
@@ -0,0 +1,51 @@
+{- |
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+
+-}
+module Kore.Step.Simplification.Conditional
+    ( simplifyPredicate
+    ) where
+
+import qualified Control.Monad.Trans.Class as Monad.Trans
+
+import Branch
+import Kore.Internal.Pattern
+    ( Conditional (..)
+    )
+import qualified Kore.Internal.Pattern as Pattern
+import Kore.Logger
+    ( LogMessage
+    , WithLog
+    )
+import qualified Kore.Step.Condition.Evaluator as Predicate
+    ( simplify
+    )
+import Kore.Step.Simplification.Simplify
+    ( MonadSimplify
+    , SimplifierVariable
+    )
+import Kore.Step.Substitution
+    ( mergePredicatesAndSubstitutions
+    )
+
+{-| Simplifies the predicate inside a 'Conditional'.
+-}
+simplifyPredicate
+    ::  ( SimplifierVariable variable
+        , MonadSimplify simplifier
+        , WithLog LogMessage simplifier
+        )
+    => Conditional variable term
+    -- ^ Contains the condition to be evaluated.
+    -> BranchT simplifier (Conditional variable term)
+simplifyPredicate Conditional {term, predicate, substitution} = do
```

Comment:
Could you explain why you defined this function and replaced `Kore.Step.Simplification.Pattern.simplifyPredicate`? They seem to be the same function

Reply:
The signature changed. A `Pattern` is a `Conditional` whose term is a `TermLike`, so, when I moved to a generic `term`, I thought I should move the function to a file called `Conditional`.